src/stores/Research/research.store.tsx

```diff
@@ -297,15 +321,15 @@ export class ResearchStore extends ModuleStore {
   }
 
   public async moderateResearch(research: IResearch.ItemDB) {
-    if (!hasAdminRights(toJS(this.activeUser))) {
+    if (!hasAdminRights(toJS(this.activeUser as IUser))) {
```

Comment:
i try to avoid casting whenever possible; since the type of `this.activeUser` is `IUserPPDB | null | undefined`, i think a preferable approach would be 
```suggestion
    if (!this.activeUser || !hasAdminRights(toJS(this.activeUser))) {
```
& would say the same about the other user casts you added to this file